go/cli/mcap/cmd/info.go

```diff
@@ -96,7 +99,11 @@ func printInfo(w io.Writer, info *mcap.Info) error {
 		if info.Statistics != nil {
 			row = append(row, fmt.Sprintf("%*d msgs (%.2f Hz)", maxCountWidth, channelMessageCount, frequency))
 		}
-		row = append(row, fmt.Sprintf(" : %s [%s]", schema.Name, schema.Encoding))
+		if schema != nil {
+			row = append(row, fmt.Sprintf(" : %s [%s]", schema.Name, schema.Encoding))
+		} else {
+			row = append(row, " : <no schema>")
+		}
```

Comment:
You'll want to test to verify, but I think you only need the change from 102-106 to get the behavior you want. If there is no schema for the channel's schema ID, then schema is going to be nil when you look it up in the map, even without the guard on 91.

If for some reason someone put a schema into the file with ID 0, which would be a spec violation, the code as written here would obscure that schema rather than display it. I think it would probably be better to display it in that case, and instead warn in doctor (which I think we do).

Reply:
>If for some reason someone put a schema into the file with ID 0, which would be a spec violation, the code as written here would obscure that schema rather than display it. I think it would probably be better to display it in that case, and instead warn in doctor (which I think we do).

When I was writing this I thought about it the opposite way: if someone makes a channel with schema id 0, the spec says that's a schemaless channel, so we shouldn't display the schema even if there happens to be one in the map. I can see this argument too though.

Reply:
I see.

IMO, if "doctoring" is printing warnings about spec violations like the doctor tool does, I think the info command shouldn't do it - otherwise we would want to make it robust to as many kinds of badly-formed files as possible, to the point that the "doctor" command would be redundant - it's not really clear where we would want to draw the line.

I think the point of the "info" and "list schemas" commands is mainly introspection of the file contents, so IMO it's not helpful for them to make aspects of the file contents inaccessible to the user. If the user has recorded a file with zero-ID schemas, and the doctor command warns/errors on it, ideally they would be able to use the CLI subcommands to figure out exactly what the problem is, rather than needing to go parse the file manually. Doctor currently will only give them this to go on: https://github.com/foxglove/mcap/blob/main/go/cli/mcap/cmd/doctor.go#L149,L151.

We could make that a bit clearer admittedly, but currently the only way they would be able to get information about their bad schema would be through the `mcap info` or `mcap list schemas` commands. If we eagerly replace the schemas with `<no schema>` in those cases, seems like we are making life more difficult to me.

Reply:
Ok, I reverted part of this to allow schemas with id 0. I also adjusted the output so it distinguishes id 0 ("no schema") from another nonzero id that was not found ("missing schema").